### PR TITLE
Deprecate task condition "source_input" block for new "module_input" block

### PR DESCRIPTION
**Pull request by lornasong**
Closes: https://github.com/hashicorp/consul-terraform-sync/issues/567

PR containing user-facing changes to rename `source_input` to `module_input` & tests

Example:
```diff
task {
  name   = "task_a"
  module = "path/to/module"
  condition "schedule" {
    cron = "* * * * Mon"
  }
- source_input "consul-kv" {
+ module_input "consul-kv" {
    path = "my/path"
  }
}
```

Future, separate PR:
- To rename backend instances of "source_input" and SourceInput e.g. class names, file names, templating, driver
- Revisit deprecation warning language

### CODE REVIEW

File: config/task.go [hunk 1]
```diff
@@ -102,8 +104,11 @@ func (c *TaskConfig) Copy() *TaskConfig {
 	o.Module = StringCopy(c.Module)
 	o.DeprecatedSource = StringCopy(c.DeprecatedSource)
 
-	if !isSourceInputNil(c.SourceInput) {
-		o.SourceInput = c.SourceInput.Copy()
+	if !isSourceInputNil(c.ModuleInput) {
```
Comment (wilkermichael) [inline config/task.go:107]: Should isSourceInputNil be renamed to `isModuleInputNil`?

Comment (wilkermichael) [reply config/task.go:107]: Nevermind, this is covered in your next PR

File: api/representation.go [hunk 2]
```diff
@@ -41,9 +41,9 @@ func (tr taskRequest) ToTaskConfig() (config.TaskConfig, error) {
 			if tr.SourceInput.Services.Names != nil {
 				si.Names = *tr.SourceInput.Services.Names
 			}
-			tc.SourceInput = si
+			tc.ModuleInput = si
```
Comment (wilkermichael) [inline api/representation.go:44]: consider renaming instances of `si` variable to `mi` or other name that reflects `ModuleInput`

also suggest doing this for https://github.com/hashicorp/consul-terraform-sync/pull/592

**Review by wilkermichael — APPROVED** [review]

**Comment by wilkermichael** [comment]
@lornasong review complete, great work

I had to go back and forth with https://github.com/hashicorp/consul-terraform-sync/pull/592 in some instances to see if changes were made later, so got a bit confused here and there. I understand that it's a really large change though, so probably not much that could be done without creating an even more massive PR.

File: api/representation.go (hunk 2 above)
Comment (lornasong) [reply api/representation.go:44]: I can make this change in the backend PR since this is out-of-scope for this PR

**Comment by lornasong** [comment]
Thanks for the review @wilkermichael

Noted. Apologies for the confusion. I had intended from the description of this PR to be only user-facing changes and tests, and the other PR for non-user-facing changes. I was hoping that this separation would allow the reviewer for this PR to not feel that they had to worry about the non-user-facing changes, which I see didn't work

**Comment by wilkermichael** [comment]
> Thanks for the review @wilkermichael
> 
> Noted. Apologies for the confusion. I had intended from the description of this PR to be only user-facing changes and tests, and the other PR for non-user-facing changes. I was hoping that this separation would allow the reviewer for this PR to not feel that they had to worry about the non-user-facing changes, which I see didn't work

No problem, it's a hard solution

**Pull request merged**